lib/color_guard/middleware.rb

```diff
@@ -11,15 +12,37 @@ def initialize(app, config = {})
     end
 
     def call(env)
-      Thread.current[:color_guard_store] = store(Rack::Request.new(env))
-      response = @app.call(env)
-      Thread.current[:color_guard_store] = nil
+      @req = Rack::Request.new(env)
+      Thread.current[:color_guard_store] = store(@req)
 
-      response
+      if @req.path_info =~ /color_guard\/write_feature/
+        write_feature
+      elsif @req.path_info =~ /color_guard/
+        flags_list
+      else
+        response = @app.call(env)
+        Thread.current[:color_guard_store] = nil
+        response
```

Comment:
I think this method should `nil` out the `:color_guard_store` thread-local variable even if it bypasses the call through to the app